backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Application.java

```diff
@@ -437,35 +438,63 @@ public Lwjgl3Window newWindow (ApplicationListener listener, Lwjgl3WindowConfigu
 		Lwjgl3ApplicationConfiguration appConfig = Lwjgl3ApplicationConfiguration.copy(this.config);
 		appConfig.setWindowConfiguration(config);
 		if (appConfig.title == null) appConfig.title = listener.getClass().getSimpleName();
-		return createWindow(appConfig, listener, windows.get(0).getWindowHandle());
+		return createWindow(appConfig, listener, windows.get(0).getWindowHandle(), true);
+	}
+
+	/**
+	 * Creates a new {@link Lwjgl3Window} using the already existing {@link Lwjgl3ApplicationConfiguration}
+	 * and {@link ApplicationListener} from the current window and replaces it. Mostly used when
+	 * some settings require the window to be created from scratch, such as MSAA sampling. */
+	public Lwjgl3Window recreateWindow() {
+		return recreateWindow(config);
+	}
+
+	private Lwjgl3Window recreateWindow(Lwjgl3ApplicationConfiguration appConfig) {
+		//Some settings don't work, such as disableAudio, which would require the
+		//Gdx.audio object to be recreated. Changing window size this way is also not recommended
+		//as it can lead to really odd values being passed into the ApplicationListener.resize() method.
+		config.set(appConfig);
```

Comment:
If some settings don't work, that really, really needs to be tested and documented. Tested because we need to know what settings could be broken, and documented so users are aware that behavior is different for recreated windows.

Reply:
>we can't expect users to source-dive to find out that disableAudio doesn't work anymore. We really need to know what configuration works, doesn't work as expected, or doesn't work at all when a window is recreated.

I should've been clearer with what I meant when explaining how some settings don't work. I brought up the example of `config.disableAudio=true` not working because that requires the `Gdx.audio` object to be reinstated and it really wasn't worth the hassle. But there's no way for the user to change that setting, the only place where `recreateWindow()` is used is in `Lwjgl3Graphics`, which uses the same `Lwjgl3ApplicationConfiguration` the programmer creates in the `DesktopLauncher` - there's no risk of anything other than the back buffer config settings getting changed.


The fact that some settings get kinda wonky when recreating the window is exactly why the main `recreateWindow()` function is private: to stop the programmer from modifying settings he really shouldn't be able to. I talked to others and the consensus was that there really is no reason for the user to be able to modify settings such as `maxNetThreads` or `audioDeviceSimultaneousSources`, but MSAA samples does make sense. If the user wants to change the window's size, he should absolutely not recreate the window, because that leads to odd results with the `width` and `height` values passes in the `ApplicationListener.resize()` method. But again, there's no possible way the programmer can change this outside of `DesktopLauncher`.